ui/helpers/formatters.ts

```diff
@@ -0,0 +1,117 @@
+import { useSelector } from 'react-redux';
+import { getResolvedLocale } from '../ducks/locale/locale';
+
+// TODO: Move this to @metamask/core
+
+const twoDecimals = {
+  minimumFractionDigits: 2,
+  maximumFractionDigits: 2,
+};
+
+const threeDecimals = {
+  minimumFractionDigits: 3,
+  maximumFractionDigits: 3,
+};
+
+const compactTwoDecimals: Intl.NumberFormatOptions = {
+  notation: 'compact',
+  minimumFractionDigits: 2,
+  maximumFractionDigits: 2,
+};
+
+const numberFormatCache = new Map<string, Intl.NumberFormat>();
+
+function getCachedNumberFormat(
+  locale: string,
+  options: Intl.NumberFormatOptions = {},
+) {
+  const key = `${locale}_${JSON.stringify(options)}`;
+
+  if (!numberFormatCache.has(key)) {
+    try {
+      numberFormatCache.set(key, new Intl.NumberFormat(locale, options));
+    } catch (error) {
+      if (error instanceof RangeError) {
+        // Fallback for invalid options (e.g. currency code)
+        const { style, currency, ...rest } = options;
+        numberFormatCache.set(key, new Intl.NumberFormat(locale, rest));
+      } else {
+        throw error;
+      }
+    }
+  }
+
+  // Do we need to update TypeScript ESLint to prevent this false positive?
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  return numberFormatCache.get(key)!;
+}
+
+function formatCurrency(
+  config: { locale: string },
+  value: number | bigint | Intl.StringNumericLiteral,
+  currency: Intl.NumberFormatOptions['currency'],
+  options: Intl.NumberFormatOptions = {},
+) {
+  if (!Number.isFinite(Number(value))) {
+    return '';
+  }
+
+  const numberFormat = getCachedNumberFormat(config.locale, {
+    style: 'currency',
+    currency,
+    ...options,
+  });
+  return numberFormat.format(value);
+}
+
+function formatCurrencyWithThreshold(
+  config: { locale: string },
+  value: number | bigint | Intl.StringNumericLiteral,
+  currency: Intl.NumberFormatOptions['currency'],
+) {
+  if (!Number.isFinite(Number(value))) {
+    return '';
+  }
+
+  const number = Number(value);
+  let options: Intl.NumberFormatOptions = {};
+
+  if (number === 0) {
+    return formatCurrency(config, 0, currency, twoDecimals);
+  } else if (number < 0.001) {
+    const formatted = formatCurrency(config, 0.001, currency, threeDecimals);
+    return `<${formatted}`;
+  } else if (number < 0.01) {
+    const formatted = formatCurrency(config, 0.01, currency, twoDecimals);
+    return `<${formatted}`;
+  } else if (number < 1) {
+    options = threeDecimals;
+  } else if (number < 1e6) {
+    options = twoDecimals;
+  } else if (number < 1e15) {
+    options = compactTwoDecimals;
+  }
+
+  return formatCurrency(config, number, currency, options);
+}
+
+export function createFormatters({ locale }: { locale: string }) {
```

Comment:
Main point of interest: 
- createFormatters is a "core" function that will return pre-defined formatting functions
- in React, `useFormatters` will return locale-aware versions of these functions

Reply:
I like this approach. My only thought here is that we won't be able to use `useFormatters` outside of a react component since it's a hook.

Question: In this case (say a utility function outside of a component), would be pass the formatter into the utility? Or would we call `createFormatters`directly?

Reply:
Good question, yes createFormatters is the core and provided for non-component use

Similar to popular libs were there's a core, and React/Vue/Svelte wrappers added on top